### PR TITLE
Use Awaitility to instead of Thread sheep method.

### PR DESCRIPTION
**Pull request by mattisonchao**
- Use Awaitility to avoid busy waiting.
- Refactor some code to Java8

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
LGTM

we are using a lot Awaiatility in Apache Pulsar.

would you like to follow up and enhance more tests ?

**Comment by eolivelli** [comment]
@tisonkun @Randgalt @cammckenzie 
if you have time please take a look, the change is simple

I like Awaitility, we started to use it in Pulsar in order to make tests code cleaner and in general reduce flakyness

**Comment by mattisonchao** [comment]
> LGTM
> 
> we are using a lot Awaiatility in Apache Pulsar.
> 
> would you like to follow up and enhance more tests ?

Sure, I will do that.

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your work!

I'd suggest separate a commit to refactor lambda expression for your next contributions @mattisonchao

File: curator-framework/src/test/java/org/apache/curator/framework/imps/TestCleanState.java [hunk 1]
```diff
@@ -43,10 +44,8 @@ public static void closeAndTestClean(CuratorFramework client)
             EnsembleTracker ensembleTracker = internalClient.getEnsembleTracker();
             if ( ensembleTracker != null )
             {
-                while ( ensembleTracker.hasOutstanding() )
-                {
-                    Thread.sleep(100);
-                }
+                Awaitility.await()
```
Comment (eolivelli) [inline curator-framework/src/test/java/org/apache/curator/framework/imps/TestCleanState.java:47]: here you are adding an implicit timeout of 10 seconds
I am fine with it, as far as it does not make the test become flaky

**Review by eolivelli — APPROVED** [review]

**Comment by eolivelli** [comment]
@mattisonchao I have merged this PR.

but in Curator we use JIRA to track the work

can you please create a ticket here and link it to this PR ?
https://issues.apache.org/jira/secure/Dashboard.jspa

Usually we require that the commit message reports the JIRA ticket ID.


@tisonkun I was too quick in using the Git Hub merge button :( 
we know have a script in Curator that prevents us from doing this kind of errors while committing PRs

**Pull request merged**